Do not allow more than one PatientName value

mrbean-bremen opened the pull request:
- for some reason, 3 values had been allowed

This is a minor change, I noticed this accidentally while looking at the decoding issue. I did not add an entry in the release notes, let me know if one is needed.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation - not needed
- [x] I have included unit tests
- [ ] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

mrbean-bremen committed:
Do not allow more than one PatientName value

- for some reason, 3 values had been allowed

mrbean-bremen commented:
Apparantly there are tests that rely on PatientName having more than one value...

gofal commented:
The limitation is because of DICOM standard: https://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html

PN is defined as:

> A character string encoded using a 5 component convention. The character code 5CH (the BACKSLASH "\" in ISO-IR 6) shall not be present, as it is used as the delimiter between Values in multi-valued Data Elements. The string may be padded with trailing spaces. For human use, the five components in their order of occurrence are: family name complex, given name complex, middle name, name prefix, name suffix.
> 
> Note
> HL7 prohibits leading spaces within a component; DICOM allows leading and trailing spaces and considers them insignificant.
> 
> Any of the five components may be an empty string. The component delimiter shall be the caret "^" character (5EH). There shall be no more than four component delimiters, i.e., none after the last component if all components are present. Delimiters are required for interior null components. Trailing null components and their delimiters may be omitted. Multiple entries are permitted in each component and are encoded as natural text strings, in the format preferred by the named person.
> 
> For veterinary use, the first two of the five components in their order of occurrence are: responsible party family name or responsible organization name, patient name. The remaining components are not used and shall not be present.
> 
> This group of five components is referred to as a Person Name component group.
> 
> For the purpose of writing names in ideographic characters and in phonetic characters, up to 3 groups of components (see [Annex H](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/chapter_H.html), [Annex I](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/chapter_I.html) and [Annex J](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/chapter_J.html)) may be used. The delimiter for component groups shall be the equals character "=" (3DH). There shall be no more than two component group delimiters, i.e., none after the last component group if all component groups are present. The three component groups of components in their order of occurrence are: an alphabetic representation, an ideographic representation, and a phonetic representation.
> 
> Any component group may be absent, including the first component group. In this case, the person name may start with one or more "=" delimiters. Delimiters are required for interior null component groups. Trailing null component groups and their delimiters may be omitted.
> 
> Precise semantics are defined for each component group. See [Section 6.2.1.2](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html#sect_6.2.1.2).
> 
> For examples and notes, see [Section 6.2.1.1](https://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html#sect_6.2.1.1).

The patientname has up to 5 components, and one Dicom-Tag may have up to 3 groups of such components.
So the PN "Doe^Jane=Doe^Jane=Doe^Jane=Doe^Jane" seems to be invalid regarding to the standard?

mrbean-bremen commented:
> The limitation is because of DICOM standard

No, this was about the VM, not about the number of components and component groups.
The number of components (maximum 5) and component groups (maximum 3) is checked elsewhere, and as you can see I also added a test that checks that, because there was none.

The superfluous check was specifically for the number of values for `PatientName`, which has VM 1, and not for any tag with VR `PN` . I don't know how the check got there, but I suspect it was a "fix" to make that serialization test pass that I changed to create valid DICOM (I guess before your time).

gofal commented:
Ah, now i got the point, thanks for explaining :-)
